src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -69,5 +81,40 @@ private static string GetPropertyName<TSourceType>(Expression<Func<TSourceType,
 
             throw new NotSupportedException($"Unsupported type of expression: {expression.GetType().Name}");
         }
+
+        /// <summary>
+        /// Constructs a lambda expression for a field resolver to return the specified query argument
+        /// from the resolve context.
+        /// </summary>
+        internal static LambdaExpression GetParameterExpression(Type parameterType, QueryArgument queryArgument)
+        {
+            //construct a call to ResolveFieldContextExtensions.GetArgument, passing in the appropriate default value
+            var getArgumentMethodTyped = _getArgumentMethod.MakeGenericMethod(parameterType);
+            var resolveFieldContextParameter = Expression.Parameter(typeof(IResolveFieldContext), "context");
+            var queryArgumentExpression = Expression.Constant(queryArgument, typeof(QueryArgument));
+            //e.g. Func<IResolveFieldContext, int> = (context) => ResolveFieldContextExtensions.GetArgument<int>(context, queryArgument.Name, queryArgument.DefaultValue);
```

Comment:
I see no default value here.

Reply:
Yeah the xml comments need to be updated. 

This code was tricky to get working properly because attributes can change the query argument properties after the expression is constructed. So now it references the query argument directly.

Reply:
So even if the name or default value changes via attributes, the behavior is correct.

Reply:
XML comments have been updated